python/ray/util/data/dataset.py

```diff
@@ -16,14 +16,35 @@ class MLDataset(ParallelIterator[pd.DataFrame]):
         batch_size (int): The batch size of the current dataset. It should be
             larger than zero, and 0 means unknown.
     """
-
     def __init__(self, actor_sets: List["_ActorSet"], name: str,
                  parent_iterators: List[ParallelIterator[pd.DataFrame]],
                  batch_size: int, repeated: bool):
         super(MLDataset, self).__init__(actor_sets, name, parent_iterators)
         self._batch_size = batch_size
         self._repeated = repeated
 
+    @classmethod
+    def from_modin(cls, df, num_shards: int = 2):
+        """Create a MLDataset from a Modin Dataframe.
+
+        Args:
+            df (modin.pandas.DataFrame): A Modin Dataframe.
+            num_shards (int): The number of worker actors to create.
+        """
+        try:
+            import modin.pandas as pd
+        except ImportError:
+            raise ImportError("Cannot convert from Modin because "
+                              "Modin is not installed.")
```

Comment:
```suggestion
        except ImportError:
            raise ImportError("Cannot convert from Modin because "
                              "Modin is not installed.") from None
```